### PR TITLE
Apply white list

### PR DESCRIPTION
**Pull request by v-limliu**
# Type of change? 

- [ ] Bug fix
- [x] New feature
- [ ] Enhancement
- [ ] Breaking change (breaks backward compatibility)

# Description, Context, Motivation 

Cache module has added more unexpected tag ,properties.So need a white-list to filter these ones.

This PR will apply white-list to tag or property names retrieved from IoTHub while rebuilding the device group filter cache

**Checklist:**

- [x] All tests passed
- [x] The code follows the code style and conventions of this project
- [ ] The change requires a change to the documentation
- [ ] I have updated the documentation accordingly

# Motivation for the change

It is a new feature

### CODE REVIEW

**Comment by zhang-hua** [comment]
This PR is relevant to another PR [#47](https://github.com/Azure/pcs-storage-adapter-java/pull/47) in storage adapter to use "ETag" instead of "Etag" in the ValueApiModel.

File: app/com/microsoft/azure/iotsolutions/uiconfig/services/Cache.java [hunk 1]
```diff
@@ -49,61 +54,42 @@ public Cache(IStorageAdapterClient storageClient,
...
+            log.info(String.format("%s:%s not found.", CacheCollectionId, CacheKey), ex);
+            return CompletableFuture.completedFuture(new CacheValue(new HashSet<>(), new HashSet<>()));
         }
     }
 
     @Override
-    public CompletionStage<CacheValue> SetCacheAsync(CacheValue cache) throws BaseException {
-        if (cache.getReported() == null) {
-            cache.setReported(new HashSet<String>());
+    public CompletionStage<CacheValue> setCacheAsync(CacheValue cacheValuesToAdd) throws BaseException {
```
Comment (timlaverty) [inline app/com/microsoft/azure/iotsolutions/uiconfig/services/Cache.java]: Any time the cache gets rebuilt this is called.

File: app/com/microsoft/azure/iotsolutions/uiconfig/services/Cache.java [hunk 2]
```diff
@@ -49,61 +54,42 @@ public Cache(IStorageAdapterClient storageClient,
...
-            ValueApiModel model = null;
-            try {
-                model = this.storageClient.getAsync(CacheCollectionId, CacheKey).toCompletableFuture().get();
-            } catch (ResourceNotFoundException e) {
-                log.info(String.format("SetCacheAsync %s:%s not found.", CacheCollectionId, CacheKey));
-            } catch (InterruptedException | ExecutionException e) {
-                log.error(String.format("SetCacheAsync InterruptedException occured in storageClient.getAsync(%s, %s).", CacheCollectionId, CacheKey));
-                throw new ExternalDependencyException("SetCacheAsync failed");
-            }
+            ValueApiModel model = this.getCurrentCacheFromStorage();
```
Comment (timlaverty) [inline app/com/microsoft/azure/iotsolutions/uiconfig/services/Cache.java]: call to docdb to get current cache

File: app/com/microsoft/azure/iotsolutions/uiconfig/services/Cache.java [hunk 3]
```diff
@@ -49,61 +54,42 @@ public Cache(IStorageAdapterClient storageClient,
...
-                    return CompletableFuture.supplyAsync(() -> cache);
+                CacheValue cacheServer = Json.fromJson(Json.parse(model.getData()), CacheValue.class);
+                this.updateCacheValues(model, cacheValuesToAdd);
+                if (cacheValuesToAdd.getTags().size() == cacheServer.getTags().size() && cacheValuesToAdd.getReported().size() == cacheServer.getReported().size()) {
+                    return CompletableFuture.completedFuture(cacheValuesToAdd);
                 }
             }
 
-            String value = Json.stringify(Json.toJson(cache));
+            String value = Json.stringify(Json.toJson(cacheValuesToAdd));
```
Comment (timlaverty) [inline app/com/microsoft/azure/iotsolutions/uiconfig/services/Cache.java]: new values added here

File: app/com/microsoft/azure/iotsolutions/uiconfig/services/external/ValueApiModel.java [hunk 4]
```diff
@@ -41,7 +41,7 @@ public void setData(String data) {
         this.data = data;
     }
 
-    @JsonProperty("Etag")
+    @JsonProperty("ETag")
```
Comment (timlaverty) [inline app/com/microsoft/azure/iotsolutions/uiconfig/services/external/ValueApiModel.java]: this is the service level, it doesn't change the REST api

**Pull request merged**